crate/operator/expand_volume.py

```diff
@@ -0,0 +1,507 @@
+                    operation=WebhookOperation.UPDATE,
+                )
+
+                await check_all_data_nodes_present(
+                    conn_factory,
+                    old_replicas,
+                    new_replicas,
+                    f"data-{node_name}",
+                    logger,
+                )
+            elif old_replicas > new_replicas:
+                # suspend the cluster -> scale down to 0 replicas
+                # First check if the cluster is healthy at all,
+                # and prevent scaling down if not.
+                await _ensure_cluster_healthy(
+                    name, namespace, apps, conn_factory, logger
+                )
+                index, *_ = field_path
+                index = int(index)
+                node_spec = spec["nodes"]["data"][index]
+                node_name = node_spec["name"]
+                sts_name = f"crate-data-{node_name}-{name}"
+                statefulset = await apps.read_namespaced_stateful_set(
+                    namespace=namespace, name=sts_name
+                )
+                current_replicas = statefulset.spec.replicas
+                if current_replicas != new_replicas:
+                    await update_statefulset(
+                        apps, namespace, sts_name, statefulset, new_replicas
+                    )
+                await send_operation_progress_notification(
+                    namespace=namespace,
+                    name=name,
+                    message="Suspending cluster and waiting for Persistent "
+                    "Volume Claim(s) to be resized.",
+                    logger=logger,
+                    status=WebhookStatus.IN_PROGRESS,
+                    operation=WebhookOperation.UPDATE,
+                )
+
+                await check_all_data_nodes_gone(core, namespace, name, old)
+
+
+async def expand_volume(
+    core: CoreV1Api,
+    namespace: str,
+    name: str,
+    data_diff_items: kopf.Diff,
+    logger: logging.Logger,
+):
+    """
+    Expand a cluster's disk size according to the given ``data_diff_items``.
+
+    :param core: An instance of the Kubernetes Core V1 API.
+    :param namespace: The Kubernetes namespace for the CrateDB cluster.
+    :param name: The CrateDB custom resource name defining the CrateDB cluster.
+    :param old: The old resource body.
+    :param data_diff_items: A list of changes made to the individual
+        data node specifications.
+    """
+    all_pvcs = await get_pvcs_in_namespace(core, namespace, name, "hot")
+    pvc_storage = {}
+
+    await send_operation_progress_notification(
+        namespace=namespace,
+        name=name,
+        message="Suspending cluster and waiting for Persistent "
+        "Volume Claim(s) to be resized.",
+        logger=logger,
+        status=WebhookStatus.IN_PROGRESS,
+        operation=WebhookOperation.UPDATE,
+    )
+
+    for pvc in all_pvcs:
+        if not pvc["name"].startswith("data0"):
```

Comment:
you are right. it comes from here https://github.com/crate/crate-operator/blob/master/crate/operator/create.py#L463 so depending on the number of disks per node it could also be `data1` `data2` etc. will move this into a constant and ditch the check for the index.